### DISCUSSION TITLE
Moving From Database To Sqs Queue connection

### DISCUSSION THREAD

**Discussion by emadrashadmuhammed** [laravel/framework#52885] [Q&A] [answered]
I was wondering what will happen to all queued jobs in my jobs table when my connection was database and not yet processed by the daemon as i am moving to sqs queue connection and when i changed my env file to the sqs connection information all new jobs will use the new queue but what about the older ones will it remains in jobs table and will not be processed thanks ...

**Comment by henzeb** [comment]
That won't be done automatically. you have to create a legacy worker yourself that processes jobs from database connection while new ones get pushed to sqs and processed by a worker that uses the sqs connection. when all database jobs processed, you can remove the legacy worker from your configuration.

**Reply by emadrashadmuhammed** [reply]
@henzeb  thanks but what will be the configuration of the legacy worker for your information am using laravel forge and my worker config is like this command=php7.2 /home/forge/my-site.example.com/artisan queue:work --tries=1 --timeout=9000

**Reply by ghost** [reply, marked as answer]
you have the queue option on the command for the worker

    protected $signature = 'queue:work
                            {connection? : The name of the queue connection to work}
                            {--name=default : The name of the worker}
                            {--queue= : The names of the queues to work}
                            {--daemon : Run the worker in daemon mode (Deprecated)}
                            {--once : Only process the next job on the queue}
                            {--stop-when-empty : Stop when the queue is empty}
                            {--delay=0 : The number of seconds to delay failed jobs (Deprecated)}
                            {--backoff=0 : The number of seconds to wait before retrying a job that encountered an uncaught exception}
                            {--max-jobs=0 : The number of jobs to process before stopping}
                            {--max-time=0 : The maximum number of seconds the worker should run}
                            {--force : Force the worker to run even in maintenance mode}
                            {--memory=128 : The memory limit in megabytes}
                            {--sleep=3 : Number of seconds to sleep when no job is available}
                            {--rest=0 : Number of seconds to rest between jobs}
                            {--timeout=60 : The number of seconds a child process can run}
                            {--tries=1 : Number of times to attempt a job before logging it failed}';

**Reply by emadrashadmuhammed** [reply]
@marius-mcp  thanks a lot , i wrote php artisan queue:work --help but i didn't see any connection option but u really helped me so passing connection as an argument processed the parked jobs in my jobs table php artisan queue:work database

**Comment by emadrashadmuhammed** [comment]
there is no option to pass the connection to the queue:work command but i can pass the connection directly
